### PR TITLE
Refactor Navigator Card Body for global event listeners

### PR DESCRIPTION
**Pull request by jelly**
This PR aims to move the ContextMenu into the NavigatorCardBody as it has no place in `app.jsx` plus remove the context menu related event listeners to the NavigatorCardBody and stop having a per Item event listener for the Context Menu.

This is now so far all preparation work, I have a patch to make it use on `onContextMenu` but that requires more work to handle the folder switching breaks contextMenu bug.

### CODE REVIEW

File: src/navigator-card-body.jsx [hunk 1]
```diff
@@ -75,7 +82,102 @@ const compare = (sortBy) => {
...
+// eslint-disable-next-line max-len
+const ContextMenuItems = ({ path, currentDir, selected, setSelected, setHistory, setHistoryIndex, addAlert, clipboard, setClipboard, files }) => {
+    const Dialogs = useDialogs();
+
+    const _createDirectory = () => createDirectory(Dialogs, currentDir);
+    const _createLink = () => createLink(Dialogs, currentDir, files, selected[0]);
+    const _copyItem = () => {
+        copyItem(setClipboard, selected.length > 1
+            ? selected.map(s => currentDir + s.name)
+            : [currentDir + selected[0].name]);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:94]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 2]
```diff
@@ -75,7 +82,102 @@
...
+    };
+    const _pasteItem = (targetPath, asSymlink) => pasteItem(clipboard, targetPath.join("/") + "/", asSymlink, addAlert);
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:96]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 3]
```diff
@@ -75,7 +82,102 @@
...
+                setHistory,
+                setHistoryIndex,
+                path: currentDir,
+                setSelected
+            }
+        );
+    };
+
+    if (selected.length === 0)
+        return null;
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:113]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 4]
```diff
@@ -75,7 +82,102 @@
...
+
+    const menuItems = [];
+
+    // HACK: No item selected but the current folder, this depends on
+    // `selected` contained the currently viewed directory which is plain wrong
+    // from a state perspective. `selected` should be the actually "selected"
+    // items and not the current path. Refactor this later.
+    if (selected.length === 1 && selected[0].name === path[path.length - 1]) {
+        menuItems.push(
+            { title: _("Paste"), onClick: () => _pasteItem(path, false), isDisabled: clipboard === undefined },
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:123]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 5]
```diff
@@ -75,7 +82,102 @@
...
+            {
+                title: _("Paste as symlink"),
+                onClick: () => _pasteItem(path, true),
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:126]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 6]
```diff
@@ -75,7 +82,102 @@
...
+            { title: _("Edit permissions"), onClick: _editPermissions }
+        );
+    } else if (selected.length === 1) {
+        menuItems.push(
+            { title: _("Copy"), onClick: _copyItem },
+            ...(selected[0].type === "dir")
+                ? [
+                    {
+                        title: _("Paste into directory"),
+                        onClick: () => _pasteItem([...path, selected[0].name], false),
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:142]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 7]
```diff
@@ -75,7 +82,102 @@
...
+                : [],
+            { type: "divider" },
+            { title: _("Edit permissions"), onClick: _editPermissions },
+            { title: _("Rename"), onClick: _renameItem },
+            { type: "divider" },
+            { title: _("Create link"), onClick: _createLink },
+            { type: "divider" },
+            { title: cockpit.format(_("Delete")), onClick: _deleteItem, className: "pf-m-danger" },
+        );
+    } else if (selected.length > 1) {
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:155]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/navigator-card-body.jsx [hunk 8]
```diff
@@ -222,6 +327,11 @@
         }
     };
 
+    const handleContextMenu = () => {
+        const sel = path ? path[path.length - 1] : undefined;
```
Comment (cockpituous) [inline src/navigator-card-body.jsx:331]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/sidebar.jsx [hunk 9]
```diff
@@ -315,9 +315,7 @@ const DropdownWithKebab = ({
             id: "edit-permissions",
             onClick: () => {
                 editPermissions(Dialogs, {
-                    selected: selected.length === 0
-                        ? null
-                        : selected[0],
+                    selected: selected[0] || currentDirectory,
```
Comment (cockpituous) [inline src/sidebar.jsx:318]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-237-20240201-172301-e0f83b57-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by allisonkarlitskaya — APPROVED** [review]
This is a real nice cleanup, thanks.

**Pull request merged**